turbojpeg.py

```diff
@@ -227,6 +229,76 @@ def encode(self, img_array, quality=85, pixel_format=TJPF_BGR, jpeg_subsample=TJ
         finally:
             self.__destroy(handle)
 
+    """
+    @jpeg_buf           buffer containing jpeg image bytes
+    @pixel_format       pixel format of destination image
+    @jpeg_subsample     decoding format
+    @scaling_factor     optional image scaling
+    @quality            optional image quality to be reencoded with
+    @flags
+    """
+    def scale_with_quality(self, jpeg_buf, pixel_format=TJPF_BGR, jpeg_subsample=TJSAMP_420,
+        scaling_factor=None, quality=85, flags=0):
+        """decompresstoYUV with scale factor, recompresstoYUV with quality factor"""
+        handle = self.__init_decompress()
+        try:
+            if scaling_factor is not None and \
+                scaling_factor not in self.__scaling_factors:
+                raise ValueError('supported scaling factors are ' +
+                    str(self.__scaling_factors))
+            pixel_size = [3, 3, 4, 4, 4, 4, 1, 4, 4, 4, 4, 4]
+            width = c_int()
+            height = c_int()
+            jpeg_subsample = c_int()
+            jpeg_colorspace = c_int()
+            jpeg_array = np.frombuffer(jpeg_buf, dtype=np.uint8)
+            src_addr = self.__getaddr(jpeg_array)
+            status = self.__decompress_header(
```

Comment:
Pull this code that calculates scaled_width/scaled_height out into a private method.  It can be called here and from `decode`